.github/workflows/linux.yaml

```diff
@@ -0,0 +1,185 @@
+name: Linux
+
+# To debug a job, add the following step:
+# - name: Debug with tmate
+#   uses: mxschmitt/action-tmate@v3
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths-ignore:
+      - '.github/workflows/macos.yaml'
+
+env:
+  COZY_DESKTOP_DIR: "${{ github.workspace }}/cozy-desktop"
+  COZY_STACK_STORAGE: "${{ github.workspace }}/storage"
+  COZY_URL: "http://localhost:8080"
+  COZY_PASSPHRASE: "cozy"
+  GOPROXY: "https://proxy.golang.org"
+  GO111MODULE: "on"
+  NODE_ENV: "test"
+  COZY_DESKTOP_HEARTBEAT: ${{ 1000 }}
+  DISPLAY: ":99.0"
+
+jobs:
+  unit:
+    runs-on: ubuntu-latest
+    name: Lint & Unit tests
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache node modules
+        id: cache-node-modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ./node_modules
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/yarn.lock') }}
+      - name: Setup node
+        uses: actions/setup-node@v2-beta
+        with:
+            node-version: '12.14.1'
+            check-latest: true
+      - name: Setup golang
+        uses: actions/setup-go@v2
+        with:
+            go-version: '1.14'
+      - name: Setup CouchDB
+        uses: 'cobot/couchdb-action@11e2c9f532a0eee2516ff80d92990d8081f2792c' # v4
+        with:
+            couchdb version: '2.3.1'
+      - name: Setup cozy-stack
+        uses: ./.github/actions/setup-cozy-stack
+      - name: Setup local env
+        run: |
+          mkdir -p "$COZY_DESKTOP_DIR"
+          echo "NODE_ENV=test" > "${{ github.workspace }}/.env.test"
+      - name: Install dependencies
+        if: ${{ ! steps.cache-node-modules.outputs.cache-hit }}
+        run: yarn install:all
+      - name: Start Xvfb
+        run: sudo Xvfb :99 -ac -screen 0 1024x768x8 &
+      - name: Build elm
+        run: yarn build:elm
+      - name: Lint
+        run: yarn lint
+      - name: Elm tests
+        run: yarn test:elm
+      - name: World tests
+        run: yarn test:world
+      - name: Unit tests
+        run: yarn test:unit
+
+  integration:
+    runs-on: ubuntu-latest
+    name: Integration tests
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache node modules
+        id: cache-node-modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ./node_modules
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/yarn.lock') }}
```

Comment:
You don't have the node version in the cache key, is it something we should add?

Reply:
That's a good point. We probably should include it since some dependencies need to be recompiled when changing the Node version.